admin.js

```diff
@@ -13,6 +13,35 @@ jQuery(function () {
         jQuery('.match input[id^=' + this.id.replace(/:/g, '\\:') + ']').prop('checked', this.checked);
     });
 
+    // When all single matches of a file have been checked, mark the appropriate "check all" box as checked, too.
+    jQuery('.check-single-match').click(function(){
+        var pageIdEscaped = this.id.substr(0, this.id.indexOf('#')).replace(/:/g, '\\:');
+        var checkedAll = true;
+        jQuery('.match input[id^=' + pageIdEscaped + ']').each(function(i, el) {
+            checkedAll = checkedAll && jQuery(el).prop('checked');
+            return checkedAll;
+        });
+        if (checkedAll) {
+            jQuery('#'+pageIdEscaped).prop('checked', true);
+        }
+    });
+
+    // Consolidate the list of all checked match ids into single hidden input field as
+    // a comma-separated string. This avoids problems for huge replacement sets
+    // exceeding `max_input_vars` in `php.ini`.
+    jQuery('#replace-form').submit(function() {
+      // Consolidate checked matches into a single string variable to be posted to the backend.
+      var checkedMatchesList='';
+      jQuery('.check-single-match').each(function(i, el) {
+        if (jQuery(el).prop('checked')) {
+          checkedMatchesList+=(el.id+',');
+        }
+      });
+      // cut off the last comma:
```

Comment:
Consider using map() and join() instead.

Reply:
I'm not a big fan of external libraries either. But one thing jQuery does good is selecting elements. As with example above it should be possible to get raw element array from the query with .get() and then use pure JS on it.